File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/impl/BerryFirmnessServiceImpl.java
package io.github.lishangbu.avalon.dataset.service.impl;

import io.github.lishangbu.avalon.dataset.entity.BerryFirmness;
import io.github.lishangbu.avalon.dataset.entity.BerryFirmness_;
import io.github.lishangbu.avalon.dataset.repository.BerryFirmnessRepository;
import io.github.lishangbu.avalon.dataset.service.BerryFirmnessService;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Example;
import org.springframework.data.domain.ExampleMatcher;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

/// 树果坚硬度服务实现。
@Service
@RequiredArgsConstructor
public class BerryFirmnessServiceImpl implements BerryFirmnessService {
    private final BerryFirmnessRepository berryFirmnessRepository;

    @Override
    public Page<BerryFirmness> getPageByCondition(BerryFirmness berryFirmness, Pageable pageable) {
        return berryFirmnessRepository.findAll(
                Example.of(
                        berryFirmness,
                        ExampleMatcher.matching()
                                .withMatcher(
                                        BerryFirmness_.NAME,
                                        ExampleMatcher.GenericPropertyMatchers.contains())
                                .withMatcher(
                                        BerryFirmness_.INTERNAL_NAME,
                                        ExampleMatcher.GenericPropertyMatchers.contains())
                                .withIgnoreNullValues()),
                pageable);
    }

    @Override
    public List<BerryFirmness> listByCondition(BerryFirmness berryFirmness) {
        ExampleMatcher matcher =
                ExampleMatcher.matching()
                        .withIgnoreNullValues()
                        .withMatcher(
                                BerryFirmness_.NAME,
                                ExampleMatcher.GenericPropertyMatchers.contains())
                        .withMatcher(
                                BerryFirmness_.INTERNAL_NAME,
                                ExampleMatcher.GenericPropertyMatchers.contains());
        return berryFirmnessRepository.findAll(Example.of(berryFirmness, matcher));
    }

    @Override
    public BerryFirmness save(BerryFirmness berryFirmness) {
        return berryFirmnessRepository.save(berryFirmness);
    }

    @Override
    public BerryFirmness update(BerryFirmness berryFirmness) {
        return berryFirmnessRepository.save(berryFirmness);
    }

    @Override
    public void removeById(Long id) {
        berryFirmnessRepository.deleteById(id);
    }
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/impl/BerryFirmnessServiceImpl.kt
package io.github.lishangbu.avalon.dataset.service.impl

import io.github.lishangbu.avalon.dataset.entity.dto.BerryFirmnessSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.BerryFirmnessView
import io.github.lishangbu.avalon.dataset.entity.dto.SaveBerryFirmnessInput
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateBerryFirmnessInput
import io.github.lishangbu.avalon.dataset.repository.BerryFirmnessRepository
import io.github.lishangbu.avalon.dataset.service.BerryFirmnessService
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

/** 树果硬度服务实现 */
@Service
class BerryFirmnessServiceImpl(
    /** 树果硬度仓储 */
    private val berryFirmnessRepository: BerryFirmnessRepository,
) : BerryFirmnessService {
    /** 按条件分页查询树果硬度*/
    override fun getPageByCondition(
        specification: BerryFirmnessSpecification,
        pageable: Pageable,
    ): Page<BerryFirmnessView> = berryFirmnessRepository.findAll(specification, pageable).mapRows(::BerryFirmnessView)

    /** 根据条件查询树果硬度列表 */
    override fun listByCondition(
        specification: BerryFirmnessSpecification,
    ): List<BerryFirmnessView> = berryFirmnessRepository.findAll(specification).map(::BerryFirmnessView)

    /** 保存树果硬度 */
    override fun save(
        command: SaveBerryFirmnessInput,
    ): BerryFirmnessView = BerryFirmnessView(berryFirmnessRepository.save(command.toEntity()))

    /** 更新树果硬度 */
    override fun update(command: UpdateBerryFirmnessInput): BerryFirmnessView = BerryFirmnessView(berryFirmnessRepository.save(command.toEntity()))

    /** 按 ID 删除树果硬度 */
    override fun removeById(id: Long) {
        berryFirmnessRepository.deleteById(id)
    }
}
